src/Castle.Core/DynamicProxy/InvalidProxyConstructorArgumentsException.cs

```diff
@@ -15,19 +15,33 @@
 namespace Castle.DynamicProxy
 {
 	using System;
+	using System.Runtime.Serialization;
 
-#if FEATURE_SERIALIZATION
 	[Serializable]
-#endif
 	public class InvalidProxyConstructorArgumentsException : ArgumentException
 	{
 		internal InvalidProxyConstructorArgumentsException(string message, Type proxyType, Type classToProxy) : base(message)
 		{
-			ProxyType = proxyType;
 			ClassToProxy = classToProxy;
+			ProxyType = proxyType;
+		}
+
+		private protected InvalidProxyConstructorArgumentsException(SerializationInfo info, StreamingContext context) : base(info, context)
+		{
+			// Binary serialization on .NET Core does not support `Type`,
+			// so we need to make do with type name strings:
+			ClassToProxy = Type.GetType(info.GetString("classToProxy"));
+			ProxyType = Type.GetType(info.GetString("proxyType"));
 		}
 
 		public Type ClassToProxy { get; private set; }
 		public Type ProxyType { get; private set; }
+
+		public override void GetObjectData(SerializationInfo info, StreamingContext context)
+		{
+			base.GetObjectData(info, context);
+			info.AddValue("classToProxy", ClassToProxy.AssemblyQualifiedName);
+			info.AddValue("proxyType", ProxyType.AssemblyQualifiedName);
```

Comment:
Would it be better to use `nameof` with the properties?

Reply:
My reasoning against using `nameof` was that serialized data follows its own contract that should remain as stable over time as possible (since one might persist data); if you change type or member names, that shouldn't necessarily change the contract for serialized data.

Granted, its unlikely that someone should want to persist our exception types then deserialize after many months :laughing: so I'm happy to change this to `nameof`.

Reply:
Fair enough point, happy to leave it.